_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -1027,6 +1038,28 @@ public static function jetpack_connection_test() {
 		}
 	}
 
+	/**
+	 * Get plugins connected to the Jetpack.
+	 *
+	 * @return WP_REST_Response|WP_Error Response or error object, depending on the request result.
+	 */
+	public static function get_connection_plugins() {
+		$plugins = ( new Connection_Manager() )->get_connected_plugins();
+
+		if ( is_wp_error( $plugins ) ) {
+			return $plugins;
```

Comment:
I tested a situation where `Manager::get_connected_plugins` returns a `WP_Error`. The modal just displays a spinner, and the endpoint returns a 500 status code. Maybe we should handle this a bit more gracefully. 

If the disconnect modal can't get the connected plugins for some reason, I think it should still work properly and display the other data (stats, subscribers, etc.). What do you think?

Reply:
That's a good catch, I didn't think about it.
Do you think we should notify the user that we weren't able to load the plugins and the connection may still be used by them? Or we should simply keep the "Connected Plugins" section hidden because that scenario is not very likely?

Reply:
I modified the code so the modal would still show up if the endpoint returns an error, ignoring the "Connected Plugins" section entirely. That information is supplemental after all.
Please let me know if you agree with the solution.

Reply:
Yeah, I think that makes sense. Like you said, it's supplemental info.

I tested this after the new changes, and the modal displays as expected. A 500 status code shows up in the console, and I assume that it's intentional.